src/api/routes/rosetta/construction.ts

```diff
@@ -297,5 +324,161 @@ export function createRosettaConstructionRouter(db: DataStore): RouterWithAsync
     }
   });
 
+  //construction/payloads endpoint
+  router.postAsync('/payloads', async (req, res) => {
+    const valid: ValidSchema = await rosettaValidateRequest(req.originalUrl, req.body);
+    if (!valid.valid) {
+      res.status(400).json(makeRosettaError(valid));
+      return;
+    }
+
+    const options = getOptionsFromOperations(req.body.operations);
+    if (options == null) {
+      res.status(400).json(RosettaErrors.invalidOperation);
+      return;
+    }
+
+    const amount = options.amount;
+    if (!amount) {
+      res.status(400).json(RosettaErrors.invalidAmount);
+      return;
+    }
+
+    const fees = options.fee;
+    if (!fees) {
+      res.status(400).json(RosettaErrors.invalidFees);
+      return;
+    }
+
+    const publicKeys: RosettaPublicKey[] = req.body.public_keys;
+    if (!publicKeys) {
+      res.status(400).json(RosettaErrors.emptyPublicKey);
+      return;
+    }
+
+    const recipientAddress = options.token_transfer_recipient_address;
+    if (!recipientAddress) {
+      res.status(400).json(RosettaErrors.invalidRecipient);
+      return;
+    }
+    const senderAddress = options.sender_address;
+
+    if (!senderAddress) {
+      res.status(400).json(RosettaErrors.invalidSender);
+      return;
+    }
+
+    const accountInfo = await new StacksCoreRpcClient().getAccount(senderAddress);
+
+    let tokenTransferOptions: UnsignedTokenTransferOptions | UnsignedMultiSigTokenTransferOptions;
+
+    if (publicKeys.length !== 1) {
+      //TODO support multi-sig in the future.
+      res.status(400).json(RosettaErrors.needOnePublicKey);
```

Comment:
Do other rosetta implementations do the same thing when there are more than 1 public keys?

Reply:
Other implementations that I have looked at don't require a public key to build an unsigned transaction. 
Is there a way we can create a transaction without specifying the public key?

Reply:
Here is the PR containing those changes. https://github.com/blockstack/stacks-blockchain-api/pull/288

Reply:
PR #288 merged. This PR needs a rebase to fix the conflicting file and then looks good to merge 👍